sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveTableScanExec.scala

```diff
@@ -138,27 +153,35 @@ case class HiveTableScanExec(
     }
   }
 
+  // exposed for tests
+  @transient lazy val rawPartitions = {
+    val prunedPartitions = if (sparkSession.sessionState.conf.metastorePartitionPruning) {
```

Comment:
Checked the history. It sounds like @liancheng can answer whether this is still needed or not. : )

https://github.com/apache/spark/pull/7421#issuecomment-122527391